docs/jumpstart/week-5.md

```diff
@@ -0,0 +1,623 @@
+together with `&`. Here are a some examples: `/some/path?param1=value1&param2=value2&param3=value3` and
+`/some/path?param1=value1&param1=value2&param1=value3` (for multiple values assigned to a single parameter). The way 
+that we currently have it set up, the `getRequestParameterAsString` cannot have an empty value, so if you would like to 
+include query params then you should create a new method which allows empty values.
+
+Here's an example of a dissected URL with query parameters if you're interested in seeing what it looks like.
+
+![complex-url](./img/complex_url.png)
+
+*https://doepud.co.uk/blog/anatomy-of-a-url*
+
+#### Creating a Sub-Router
+
+A subrouter is a custom router which handles routing for a group of routes with the same path prefix. For example, if 
+you have a bunch of defined routes for `/user/info/...`, then you can create a subrouter for `/user/info/`, and every
+route created in the subrouter will automatically be prefixed with that. This is not only useful because it lets you 
+type less by removing the need to copy the same prefix for similar routes, but it also allows you to separate out 
+the functionality of similar methods. For example, if you have a subrouter for something such as **_posts_** *(hint, 
+hint)*, then all functionality in that subrouter should be related to posts, keeping you more organized.
+
+To create a subrouter, it's not hard at all. All you need to do is call the `Router.mountSubRouter()` method in your
+`ApiMain` and set a few things up in your subrouter class.
+
+```java 
+// Again, in ApiMain BEFORE calling server.requestHandler().
+// Initialize a router (that was probably passed in).
+router.mountSubRouter("/path_prefix", subRouter.initializeRouter(vertx));
+
+server.requestHandler(router).listen(defaultPort);
+```
+
+And then in your subRouter class, create a new router, set up routes, and return the router.
+
+```java 
+public class SubRouter implements IRouter {
+  private final ISomethingProcessor processor;
+
+  public SubRouter(ISomethingProcessor processor) {
+    this.processor = processor;
+  }
+
+  @Override
+  public Router initializeRouter(Vertx vertx) {
+    // Create a new Router object in our subrouter.
+    Router router = Router.router(vertx);
+
+    this.registerRouteA(router);
+    this.registerRouteB(router);
+    ...
+    
+    return router;
+  }
+
+// Other methods for doing route work.
+...
+}
+```
+
+### Processor Interfaces
+
+Since the api shouldn't know anything about the database, a lot of the time processing needs to be done in the service
+directory/module. That makes things complicated though, since the api module needs to be able to call these methods 
+when a route is accessed. This can be accomplished by creating an interface for the processor in the api module and 
+implementing it in the service module. In that way, the service module is able to implement the methods api requires 
+while not needing to know exactly **how** it's implemented. 
```

Comment:
Add a sentence saying that we'll get into defining these interfaces and implementing them in the next week